backend-api/src/functions/asyncActiveSession/tokenService/tests/tokenService.test.ts

```diff
@@ -24,355 +23,16 @@ describe("Token Service", () => {
-              "https://mockJwksEndpoint.com",
-              "mockEncryptionKeyArn",
-              "mockJwe",
-              { maxAttempts: 3, delayInMillis: 1 },
-            );
-
-            expect(mockFetch).toHaveBeenCalledWith(
-              "https://mockJwksEndpoint.com",
-              {
-                method: "GET",
-              },
-            );
-            expect(result.isError).toBe(true);
-            expect(result.value).toStrictEqual({
-              errorMessage: "Unexpected network error: Error: mockError",
-              errorCategory: "SERVER_ERROR",
-            });
-          });
-        });
-
-        describe("Given the request fails with a 500 error", () => {
-          it("Returns error result", async () => {
-            mockFetch = jest.spyOn(global, "fetch").mockImplementation(() =>
-              Promise.resolve({
-                status: 500,
-                ok: false,
-                text: () => Promise.resolve("mockErrorInformaton"),
-              } as Response),
-            );
-
-            const result = await tokenService.getSubFromToken(
-              "https://mockJwksEndpoint.com",
-              "mockEncryptionKeyArn",
-              "mockJwe",
-              { maxAttempts: 3, delayInMillis: 1 },
-            );
-
-            expect(mockFetch).toHaveBeenCalledWith(
-              "https://mockJwksEndpoint.com",
-              {
-                method: "GET",
-              },
-            );
-            expect(result.isError).toBe(true);
-            expect(result.value).toStrictEqual({
-              errorMessage: "Error making http request: mockErrorInformaton",
-              errorCategory: "SERVER_ERROR",
-            });
-          });
-        });
-
-        describe("Given the response is empty", () => {
-          it("Returns an error response", async () => {
-            mockFetch = jest.spyOn(global, "fetch").mockImplementation(() =>
-              Promise.resolve({
-                status: 200,
-                ok: true,
-                headers: new Headers({
-                  header: "mockHeader",
-                }),
-                text: () => Promise.resolve(""),
-              } as Response),
-            );
-
-            const result = await tokenService.getSubFromToken(
-              "https://mockJwksEndpoint.com",
-              "mockEncryptionKeyArn",
-              "mockJwe",
-              { maxAttempts: 3, delayInMillis: 1 },
-            );
-
-            expect(mockFetch).toHaveBeenCalledWith(
-              "https://mockJwksEndpoint.com",
-              {
-                method: "GET",
-              },
-            );
-            expect(result.isError).toBe(true);
-            expect(result.value).toStrictEqual({
-              errorMessage: "Response body empty",
-              errorCategory: "SERVER_ERROR",
-            });
-          });
-        });
-
-        describe("Given the response is not valid JSON", () => {
```

Comment:
Not really required. We don't need to test JSON.parse, I don't think. If JSON.parse throws an error we will catch it and return an error response